### PR TITLE
Ad user story 25 - 26

### PR DESCRIPTION
**Pull request by adumortier**
- added the one-to-many relationship between `users` and `orders
- added a `status` column to the `Orders` table
- added the model tests of the relationships between `users` and `orders`
- made changes to the `create` action method of the `Order` controller to create the `order through the `current_user`
- added an `index` view page for `orders` and a route

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -4,6 +4,10 @@ class Order <ApplicationRecord
   has_many :item_orders
   has_many :items, through: :item_orders
 
+  belongs_to :user, optional: true
```
Comment (KellyIB) [inline app/models/order.rb]: I am not sure we should have an optional qualifier here, shouldn't orders always belong to a user?  I think in the visitor's cart, they are not allowed to checkout and create an order unless they login.

**Review by KellyIB — COMMENTED** [review]
I think we might want to remove the optional qualifier in the order model.  I think we need to update any previous tests that fail with the new relationship we have here.  I don't think an order should be able to exist without a user to create it.

File: db/schema.rb [hunk 2]
```diff
@@ -57,6 +57,9 @@
     t.integer "zip"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.bigint "user_id"
+    t.integer "status", default: 0
```
Comment (cgaddis36) [inline db/schema.rb:61]: Nice job on the relationship table, looks good

**Review by KellyIB — APPROVED** [review]
I think we might want to remove the optional qualifier in the order model.  I think we need to update any previous tests that fail with the new relationship we have here.  I don't think an order should be able to exist without a user to create it.

File: spec/features/cart/checkout_spec.rb [hunk 3]
```diff
@@ -27,7 +28,104 @@
...
+      click_button "Create Order"
+
+      expect(Order.last.name).to eq(username)
+      expect(Order.last.address).to eq(street_address)
+      expect(Order.last.city).to eq(city)
+      expect(Order.last.state).to eq(state)
+      expect(Order.last.zip).to eq(zip_code.to_i)
+      expect(Order.last.pending?).to eq(true)
+      expect(Order.last.user).to eq(User.last)
+
```
Comment (cgaddis36) [inline spec/features/cart/checkout_spec.rb:98]: Nice specificity on testing the order attributes

File: spec/features/cart/checkout_spec.rb [hunk 4]
```diff
@@ -27,7 +28,104 @@
...
+    end
   end
+  
+# User Story 26, Registered users can check out
+
+# As a registered user
+# When I add items to my cart
+# And I visit my cart
+# I see a button or link indicating that I can check out
+# And I click the button or link to check out and fill out order info and click create order
```
Comment (cgaddis36) [inline spec/features/cart/checkout_spec.rb:119]: Comments! :)

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -11,6 +11,11 @@
     it {should validate_presence_of :state}
     it {should validate_presence_of :zip_code}
   end
+
+  describe "relationships" do 
+    it {should have_many :orders}
```
Comment (cgaddis36) [inline spec/models/user_spec.rb:16]: relationships setup looks good

File: app/models/order.rb (hunk 1 above)
Comment (cgaddis36) [reply app/models/order.rb]: nice catch Kelly!

**Review by cgaddis36 — APPROVED** [review]
Everything looks good, orders for cart implementation functional and ready for merge

**Pull request merged**